vwifi.c

```diff
@@ -132,14 +186,37 @@ static void owl_scan_routine(struct work_struct *w)
  * This routine is called through workqueue, when the kernel asks to connect
  * through cfg80211_ops.
  */
+static bool is_valid_ssid(const char *connecting_ssid)
+{
+    bool is_valid = false;
+
+    const char *org_ssid_list =
+        (ssid_list && strlen(ssid_list)) ? ssid_list : default_ssid_list;
+    size_t len = strlen(org_ssid_list);
+    char *ssid_list_copy =
+        kmalloc((len + 1) * sizeof(ssid_list_copy), GFP_KERNEL);
+    strncpy(ssid_list_copy, org_ssid_list, len);
+    for (char *sdssid = strsep(&ssid_list_copy, sdssid_delim); sdssid;
+         sdssid = strsep(&ssid_list_copy, sdssid_delim)) {
+        if (!strlen(sdssid++) || strlen(sdssid) > SSID_MAX_LENGTH)
```

Comment:
`strlen` takes a pointer to a character array as a parameter, `NULL` is not a valid argument to this function. It means that you should check in advance.

Reply:
Isn't it checked by the for loop? 🤔

Reply:
Check after Line 200.

Reply:
fixed in https://github.com/sysprog21/vwifi/pull/4/commits/45ac76cdd7ce009e23fce6d6b7e3e700de08f531